conans/test/remote/scm_nested_subfolder.py

```diff
@@ -0,0 +1,38 @@
+import unittest
+from conans.test.utils.tools import TestServer, TestClient, create_local_git_repo
+from conans.test.utils.cpp_test_files import cpp_hello_conan_files
+from conans.model.ref import ConanFileReference
+from conans.client.tools.scm import Git
+import os
+from conans.util.files import save
+
+
+class SCMNestedSubfolderTest(unittest.TestCase):
+
+    # Testing as of issue #3423: https://github.com/conan-io/conan/issues/3423
+    def basic_test(self):
+        server = TestServer()
+        servers = {"default": server}
```

Comment:
servers seem unnecessary for this test

Reply:
This test is about uploading a package and installing it, how can it be done without a _server_?

Reply:
Oh, sorry, my bad, didn't see the upload.